exist-core/src/test/java/org/exist/xquery/FunctionTypeInElementContent.java

```diff
@@ -0,0 +1,174 @@
+        final String error = "err:XQTY0105 Function types are not allowed in element content. Got map(*) [at line 1, column 16, source: element test { map {} }]";
+        assertCompilationError(query, error);
+    }
+
+    /**
+     * sequence in enclosed expression with only a function type
+     */
+    @Ignore
+    @Test
+    public void sequenceOfMaps() throws XMLDBException {
+        final String query = "element test { (map {}) }";
+        final String error = "An exception occurred during query execution: err:XQTY0105 Function types are not allowed in element content. Got map(*) [source: element foo { (map{}) }]";
+        assertCompilationError(query, error);
+    }
+
+    /**
+     * there is an edge case which would evaluate to empty sequence
+     * but should arguably still throw
+     * Does still throw without location info
+     * TODO: add (sub-expression) location
+     */
+    @Test
+    public void sequenceOfMapsEdgeCase() throws XMLDBException {
+        final String query = "element test { (map {})[2] }";
+        final String error = "err:XQTY0105 Function types are not allowed in element content. Got map(*) [source: element test { (map {})[2] }]";
+        assertCompilationError(query, error);
+    }
+
+    /**
+     * -- no error is thrown at compile time
+     * TODO: add (sub-expression) location
+     */
+    @Test
+    public void ArrayOfMaps() throws XMLDBException {
+        final String query = "element test { [map {}] }";
+        final String error = "An exception occurred during query execution: err:XQTY0105 Function types are not allowed in element content. Got map(*) [source: element test { [map{}] }]";
+        assertCompilationError(query, error);
+    };
+
+    /**
+     * This could throw at compile time, but does not
+     * TODO: add (sub-expression) location
+     */
+    @Ignore
+    @Test
+    public void mapConstructorInSubExpression() throws XMLDBException {
+        final String query = "element test { \"a\", map {} }";
+        final String error = "err:XQTY0105 Function types are not allowed in element content. Got map(*) [at line 1, column 20, source: element test { map {} }]";
+        assertCompilationError(query, error);
+    }
+
+    private void assertCompilationError(final String query, final String error) throws XMLDBException {
+        final XQueryService service = (XQueryService)existEmbeddedServer.getRoot().getService("XQueryService", "1.0");
+
+        try {
+            service.compile(query);
+            fail("no XMLDBException was thrown during compilation.");
+        } catch (XMLDBException ex) {
+            assertEquals( error, ex.getMessage() );
+        }
+    }
+    private void assertCompilationSuccess(final String query) throws XMLDBException {
+        final XQueryService service = (XQueryService)existEmbeddedServer.getRoot().getService("XQueryService", "1.0");
+
+        service.compile(query);
+        assertTrue( true );
```

Comment:
This line does nothing! You probably wanted to check that the result of `service.compile` is non-null instead.

Reply:
It asserts that no compile error was thrown. I can change that to a non-null check, if needed. What is the method name?
`assertNonNull(compilationResult)`?

Reply:
It doesn't assert that there was no error... an exception not being thrown and returning void is enough for that ;-)

You can use `assertNotNull`